### ISSUE TITLE
[Feature Request]: Support default field value in cross-language BigQuery Storage Write API

### ISSUE THREAD

**Issue by Abacn** [apache/beam#28008]
### What would you like to happen?

As an example, user has field in the target BQ table called insert_timestamp filled with the current time by default, the data gets written to BQ may not contain this field.

Previously, the FILE_LOAD mode support this use case. However, using Storage Write Api (xlang implementation), the job throws an attribute error:
```
AttributeError: 'Row' object has no attribute 'insert_timestamp' [while
running 'write customer to BigQuery/Map(<lambda at
bigquery.py:2157>)-ptransform-934']
```
The error is thrown from Python side.

The task involves checking if Java Storage Write Api support default field; and port this support to Python.

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by ahmedabu98** [comment]
The SchemaTransform does a strict check between the table's schema and input schema in https://github.com/apache/beam/blob/767a2d72510bf0d6d100e80f45c74ede913ece30/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java#L396C18-L396C32.

I see this exact use-case was foreshadowed by @slilichenko in his [comment on that PR](https://github.com/apache/beam/pull/25915#issuecomment-1496102214).

**Comment by ahmedabu98** [comment]
.take-issue

**Comment by shopmatfournier** [comment]
Chiming in that we ran into this a few weeks ago.  I've hacked around it, for now, in our (java) dataflow jobs.  I've been meaning to write up an issue and go spelunking in the code to try to fix.

Our current hack: 

- we create the table using the java BigQuery library directly, not via BigQueryIO. 
  - we create a table with the `nullable` field with some `.setDefaultValueExpresion()`  
- as for the schema that Beam/BigQueryIO knows about, we strip this field from it so they never validate it 
- rows inserted into BQ via BigQueryIO have the defaultValueExpression applied  

Downsides: It has to be a `nullable` field for this to work.  This is actually a field we end up partitioning on, so it's not great; however, it's working for us.
